index.mjs

```diff
@@ -430,6 +431,31 @@ export const coroutine = function coroutine(g) {
   });
 };
 
+//           exactly :: (Integer, Parser e s a) -> Parser e s [a]
+export const exactly = function exactly(n, parser) {
```

Comment:
To match the rest of the multi argument signatures this should probably be curried.

Reply:
So that would look like...
```
const exactlyTwoLetters = exactly(2)(letter);
```

Locally I ended up creating `sepByN` and `manyN` because I needed more flexibility than what's provided by just `sepBy`, `sepBy1`, `many`, and `many1`.

```
const many3 = manyN(3);
const sepBy2 = sepByN(2);
```

From an overall API consistency, I could see dropping `sepBy1` and `many1`, and making `sepBy` and `many` behave like `exactly`. Doing this would make the zero or matching of `many` and `sepBy` more explicit.

```
const optionalLetters = many(0)(letter);
const atLeastTwoDigits = many(2)(digit);
```

I could also see keeping `sepBy` and `many` as they are, but replacing `sepBy1` and `many1` with `sepByN` and `manyN` at which point `exactly` should be renamed to `exactlyN` for consistency.

Reply:
That's a really interesting thought that I hadn't considered actually. Originally the `many`/`many1` came from parsec and I was just matching the API there, but this seems like a nice idea. I probably wouldn't use `many` as the root "generic" parser, but rather put it in some kind of meta API like `_many` - in order to avoid a major version cycle.